src/h5web/toolbar/Toolbar.tsx

```diff
@@ -1,32 +1,34 @@
-import { ReactElement, Children } from 'react';
-import { useMeasure } from '@react-hookz/web';
-import { useList } from 'react-use';
+import type { ReactElement, ReactNode } from 'react';
+import { useMap, useMeasure } from '@react-hookz/web';
 import styles from './Toolbar.module.css';
 import Separator from './Separator';
 import OverflowMenu from './OverflowMenu';
 import MeasuredControl from './MeasuredControl';
+import flattenChildren from 'react-keyed-flatten-children';
+import { isReactElement } from '../guards';
 
 interface Props {
-  children?: ReactElement | (ReactElement | undefined)[];
+  children: ReactNode;
 }
 
 function Toolbar(props: Props) {
   const { children } = props;
 
-  // Convert `children` to array, as it cannot be used directly (and remove `undefined` children)
-  const allChildren = Children.toArray(children) as ReactElement[];
+  /* Convert `children` to flat array by traversing nested arrays and fragments.
+   * (Note that `flattenChildren` guarantees stable string keys regardless of JSX logic.) */
+  const allChildren = flattenChildren(children).filter(isReactElement);
```

Comment:
The types are simpler:

- `children` is now just `ReactNode` (i.e. `ReactChild | ReactFragment | ReactPortal | boolean | null | undefined`), which brings flexibility to users of the `Toolbar` component;
- `flattenChildren` returns `ReactChild[]` (i.e. `(ReactElement | string | number)[]` -- note that `ReactFragment` is gone), which I then filter to ignore strings and numbers.

So we end with what we really want, `ReactElement[]`, in a much less convoluted way than before.